apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/RifLayoutsProcessor.java

```diff
@@ -1503,4 +1828,195 @@ private void writeDebugLogMessages() {
+           * problems with our tests.
+           */
+          StringBuilder columnDefinition = new StringBuilder();
+          columnDefinition.append("numeric");
+          if (columnLength.isPresent() || columnScale.isPresent()) {
+            columnDefinition.append('(');
+            if (columnLength.isPresent()) {
+              columnDefinition.append(columnLength.get());
+            }
+            if (columnScale.isPresent()) {
+              columnDefinition.append(", ");
+              columnDefinition.append(columnScale.get());
+            }
+            columnDefinition.append(')');
+          }
+          columnAnnotation.addMember("columnDefinition", "$S", columnDefinition.toString());
+        }
+      }
+      annotations.add(columnAnnotation.build());
+    } else {
+      annotations.add(AnnotationSpec.builder(Transient.class).build());
+    }
+
+    return annotations;
+  }
+
+  /**
+   * Selects the java field type
+   *
+   * @param type specifies the field type {@link RifColumnType}
+   * @param isColumnOptional determines if the field is optional {@link boolean}
+   * @param columnLength specifies the column length {@link Optional<Integer>}
+   * @param columnScale specifies the column scale {@link Optional<Integer>}
+   */
+  private static TypeName selectJavaFieldType(
+      RifColumnType type,
+      boolean isColumnOptional,
+      Optional<Integer> columnLength,
+      Optional<Integer> columnScale) {
+    if (type == RifColumnType.CHAR
+        && columnLength.orElse(Integer.MAX_VALUE) == 1
+        && !isColumnOptional) return TypeName.CHAR;
+    else if (type == RifColumnType.CHAR
+        && columnLength.orElse(Integer.MAX_VALUE) == 1
+        && isColumnOptional) return ClassName.get(Character.class);
+    else if (type == RifColumnType.CHAR) return ClassName.get(String.class);
+    else if (type == RifColumnType.DATE && columnLength.orElse(0) == 8)
+      return ClassName.get(LocalDate.class);
+    else if (type == RifColumnType.TIMESTAMP && columnLength.orElse(0) == 20)
+      return ClassName.get(Instant.class);
+    else if (type == RifColumnType.NUM && columnScale.orElse(Integer.MAX_VALUE) > 0)
+      return ClassName.get(BigDecimal.class);
+    else if (type == RifColumnType.NUM
+        && columnScale.orElse(Integer.MAX_VALUE) == 0
+        && !isColumnOptional) return TypeName.INT;
+    else if (type == RifColumnType.NUM
+        && columnScale.orElse(Integer.MAX_VALUE) == 0
+        && isColumnOptional) return ClassName.get(Integer.class);
+    else throw new IllegalArgumentException("Unhandled field type: " + type.name());
+  }
+
+  /**
+   * Selects the java property type
+   *
+   * @param type specifies the field type {@link RifColumnType}
+   * @param isColumnOptional determines if the field is optional {@link boolean}
+   * @param columnLength specifies the column length {@link Optional<Integer>}
+   * @param columnScale specifies the column scale {@link Optional<Integer>}
```

Comment:
Would you mind applying this suggestion?
```suggestion
   * @param columnLength specifies the column length {@link Optional<Integer>}, for numeric types this represents the total number of digits that can be stored
   * @param columnScale specifies the column scale {@link Optional<Integer>}, for numeric types this represents how many of the total digits (see `columnLength`) are to the right of the decimal point
```